tests/lib/plugin.js

```diff
@@ -19,19 +19,27 @@ const plugin = require("../..");
  * Helper function which creates CLIEngine instance with enabled/disabled autofix feature.
  * @param {string} fixtureConfigName ESLint JSON config fixture filename.
  * @param {CLIEngineOptions} [options={}] Whether to enable autofix feature.
- * @returns {CLIEngine} CLIEngine instance to execute in tests.
+ * @returns {ESLint} ESLint instance to execute in tests.
  */
 function initCLI(fixtureConfigName, options = {}) {
-    const cli = new CLIEngine({
+    if (ESLint) { // ESLint v7+
+        return new ESLint({
+            cwd: path.resolve(__dirname, "../fixtures/"),
+            ignore: false,
+            useEslintrc: false,
+            overrideConfigFile: path.resolve(__dirname, "../fixtures/", fixtureConfigName),
+            plugins: { markdown: plugin },
+            ...options
+        });
+    }
+
+    return new CLIEngine({
         cwd: path.resolve(__dirname, "../fixtures/"),
         ignore: false,
         useEslintrc: false,
         configFile: path.resolve(__dirname, "../fixtures/", fixtureConfigName),
         ...options
-    });
-
-    cli.addPlugin("markdown", plugin);
-    return cli;
```

Comment:
I suspect tests are failing because these lines were changed. (`addPlugin()` [returns `void`](https://github.com/eslint/eslint/blob/9738f8cc864d769988ccf42bb70f524444df1349/lib/cli-engine/cli-engine.js#L675-L681), not the `CLIEngine` instance.)

Reply:
Upon further investigation, looks like this is only part of it. In addition to having different method names than `CLIEngine`, `ESLint`'s methods are also `async` and return [`LintResult[]`](https://eslint.org/docs/developer-guide/nodejs-api#return-value-1) instead of a `report` object where `.results` is the array. To avoid that repetitive ternary, it should be possible to return a wrapper that adapts the `CLIEngine` instance to match the `ESLint` API for the limited purposes of our tests:

```js
return {
    async calculateConfigForFile(filename) {
        return cli.getConfigForFile(filename);
    },
    async lintFiles(files) {
        return cli.executeOnFiles(files).results;
    },
    async lintText(text, { filePath }) {
        return cli.executeOnText(text, filePath).results;
    }
};
```

Then all the tests can use the `ESLint` API but still work with `CLIEngine` in `eslint@6`:

```js
it("description", async () => {
    const results = await eslint.lintText(text, { filePath: "test.md" });
    assert.strictEqual(results.length, 1);
    // ...
});
```

- Renamed `initCli()` to `initESLint` and its return value `cli` to `eslint`.
- Used `ESLint` methods like `lintText()` instead of `CLIEngine` methods like `executeOnText()`, no ternary needed thanks to the wrapper.